src/libs/antares/study/parts/hydro/container.cpp

```diff
@@ -132,235 +132,61 @@ bool PartHydro::LoadFromFolder(Study& study, const AnyString& folder)
-                          && ret;
-                    double temp[DAYS_PER_YEAR];
-                    auto& max = area.hydro.maxPower[area.hydro.maximum];
-                    for (int i = 0; i < DAYS_PER_YEAR; i++)
-                    {
-                        temp[i] = max[i];
-                    }
-                    area.hydro.maxPower.reset(4, DAYS_PER_YEAR, true);
-                    area.hydro.maxPower.fillColumn(area.hydro.genMaxE, 24.);
-                    area.hydro.maxPower.fillColumn(area.hydro.pumpMaxP, 0.0);
-                    area.hydro.maxPower.fillColumn(area.hydro.pumpMaxE, 24.);
-                    auto& maxP = area.hydro.maxPower[area.hydro.genMaxP];
-                    for (int i = 0; i < DAYS_PER_YEAR; i++)
-                    {
-                        maxP[i] = temp[i];
-                    }
-                }
-            }
-        }
+       // GUI part patch :
+       // We need to know, when estimating the RAM required by the solver, if the current area
+       // is hydro modulable. Therefore, reading the area's daily max power at this stage is
+       // necessary.
 
-        if (study.header.version >= 620)
-        {
-            buffer.clear() << folder << SEP << "common" << SEP << "capacity" << SEP
-                           << "creditmodulations_" << area.id << '.' << study.inputExtension;
-            ret = area.hydro.creditModulation.loadFromCSVFile(
-                    buffer, 101, 2, Matrix<>::optFixedSize, &study.dataBuffer)
-                  && ret;
-        }
-        else
-        {
-            area.hydro.creditModulation.reset(101, 2, true);
-            area.hydro.creditModulation.fill(1);
-        }
-
-        if (study.header.version >= 620)
-        {
-            buffer.clear() << folder << SEP << "common" << SEP << "capacity" << SEP << "reservoir_"
-                           << area.id << '.' << study.inputExtension;
-            ret = area.hydro.reservoirLevel.loadFromCSVFile(
-                    buffer, 3, DAYS_PER_YEAR, Matrix<>::optFixedSize, &study.dataBuffer)
-                  && ret;
-        }
-        else if (study.header.version >= 390)
+        if (!study.usedByTheSolver)
         {
             bool enabledModeIsChanged = false;
             if (JIT::enabled)
             {
                 JIT::enabled = false; // Allowing to read the area's daily max power
                 enabledModeIsChanged = true;
             }
-            buffer.clear() << folder << SEP << "common" << SEP << "capacity" << SEP << "reservoir_"
-                           << area.id << '.' << study.inputExtension;
-            ret = area.hydro.reservoirLevel.loadFromCSVFile(
-                    buffer, 3, 12, Matrix<>::optFixedSize, &study.dataBuffer)
-                  && ret;
-
-            double temp[3][DAYS_PER_YEAR];
-            uint firstDayMonth[13];
-            uint daysPerMonthDecals[12];
 
-            DataSeriesHydro::AdjustMonth(study, firstDayMonth, daysPerMonthDecals);
```

Comment:
If we don't use **AdjustMonth(...)** anymore, its declaration / definition should be removed from code

Reply:
I agree !